src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/BinnedCNVDefragmenter.java

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.hellbender.tools.sv.cluster;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.tools.sv.SVCallRecord;
+import org.broadinstitute.hellbender.utils.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * CNV defragmenter for when the intervals used for coverage collection are available. The intervals are used to adjust
+ * variant padding by rounding to the nearest bin boundary.
+ */
+public class BinnedCNVDefragmenter extends CNVDefragmenter {
+
+    protected final TreeMap<GenomeLoc, Integer> genomicToBinMap;
```

Comment:
It seems sad to abandon the nice things about the current SVIntervalTree class. Maybe (but not as part of this PR) we could eventually think about porting or parameterizing the SVIntervalTree so that it could work on other kinds of intervals.